libbpf-cargo/src/btf/btf.rs

```diff
@@ -227,6 +227,50 @@ impl<'a> Btf<'a> {
         })
     }
 
+    pub fn type_default(&self, type_id: u32) -> Result<String> {
+        let stripped_type_id = self.skip_mods_and_typedefs(type_id)?;
+        let ty = self.type_by_id(stripped_type_id)?;
+
+        Ok(match ty {
+            BtfType::Void => "std::ffi::c_void::default()".to_string(),
+            BtfType::Int(t) => {
+                let width = match (t.bits + 7) / 8 {
+                    1 => "8",
+                    2 => "16",
+                    4 => "32",
+                    8 => "64",
+                    16 => "128",
+                    _ => bail!("Invalid integer width"),
+                };
+
+                if t.encoding == btf::BtfIntEncoding::Signed {
+                    format!("i{}::default()", width)
+                } else {
+                    format!("u{}::default()", width)
+                }
+            }
```

Comment:
Can't we do `format!("{}::default()", self.type_declaration()?)` here? Instead of duplicating parts of self.type_declaration()

Reply:
Not with the current field structure of a BtfInt.

IMHO a refactor could be done here to break the type_declaration and type_definition to be functions impl by the various types themselves.  I'll upload a specific 'idea' for BtfInt and we can discuss once I get it done.

Reply:
Hmm, why is it not possible? Unclear what you mean by "current field structure"

Reply:
For some reason my brain was fixated on only using the information from t to call type_definition.  I guess there is no reason not to re-use the type_id already available.

NM.  I'll fix.